src/main/java/de/dentrassi/rpm/builder/RpmMojo.java

```diff
@@ -601,6 +602,27 @@ public void setGenerateDefaultSourcePackage(final boolean generateDefaultSourceP
     @Parameter(property = "rpm.signature")
     Signature signature;
 
+    /**
+     * An optional string specifying the compression method to use for compressing the payload of the final RPM.
+     * <p>
+     * The method name must be one of the names returned by {@link PayloadCoding#values()}.
+     * <p>
+     * If no payload coding is specified, then the default RPM compression method is used.
+     */
+    @Parameter(property = "rpm.payloadCoding")
+    String payloadCoding;
+
+    /**
+     * An optional string specifying the payload flags to use for compressing the payload of the final RPM.
+     * <p>
+     * This string generally contains the compression level to use for compressing the payload.
+     * <p>
+     * If no payload flags are specified,
+     * then the default compression level for the {@link PayloadCoding payload coding} is used.
+     */
+    @Parameter(property = "rpm.payloadFlags")
+    String payloadFlags;
```

Comment:
Is this a bit complex to expose to consumers? Especially without knowing what valid strings are (i.e. "0", "1", "2") ?

Reply:
It is actually somewhat arbitrary. The docs explain it a bit. It's probably up to packager to validate it. I will make another pull request there.

Reply:
Validation of numeric value done in https://github.com/eclipse/packager/pull/74.

Reply:
On second thought, a real datatype would be better here instead of a string, and let packager do the conversion to/from the string?

Reply:
A real datatype does sound better but it still seems like a rather complex option to expose :shrug: 
@ctron Thoughts?

Reply:
So, it seems that RPM allows bogus payload flags to end up in the final RPM.

```
%define _binary_payload       w91.gzdio
rpmbuild --ba ./fedora-release.spec
```

```bash
$ rpm -q --queryformat "%{PAYLOADCOMPRESSOR} %{PAYLOADFLAGS}\n" ../RPMS/noarch/fedora-release-41-31.noarch.rpm
gzip 91
```

But, it obviously didn't use compression level 91. I think it used 9, if it's taking the first digit. We'd have to check the C code.

Reply:
Our issue is that:

1. Different compressors take different options, though some are shared: `([0-9]|[0-9]+)` for level for all, and `T([0-9])+` for threads for lzma, xz, and zstd
2. Payload flags are parsed differently based on the compressor. For example, for gzip/bzip2 only 0-9 is allowed for level, but for zstd, multiple digits are allowed.